[Azure Cognitive Search] Fix vectorSearchAlgorithmConfiguration polymorphism

Careyjmac opened the pull request:
# Data Plane API - Pull Request
Fix new Azure Cognitive Search swagger to correctly define vectorSearchAlgorithmConfiguration with polymorphism for each algorithm kind.

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [X] a private preview
- [X] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

Careyjmac committed:
Fix vectorSearchAlgorithmConfiguration polymorphism

Careyjmac committed:
Fix vectorSearchAlgorithmConfiguration polymorphism

Careyjmac committed:
Fix lintdiff

Careyjmac committed:
Fix lintdiff

robertklee reviewed (approved):
LGTM, thanks!

raych1 committed:
Updated common script part in typespec e2e doc (#23875)

* Updated common script part in typespec e2e doc

* Update documentation/typespec-end-to-end-scenarios.md

Co-authored-by: Wes Haggard <weshaggard@users.noreply.github.com>

* Update documentation/typespec-end-to-end-scenarios.md

Co-authored-by: Wes Haggard <weshaggard@users.noreply.github.com>

* Resolve comments

* Refactor the content

* Update bulletin

---------

Co-authored-by: Wes Haggard <weshaggard@users.noreply.github.com>

JainRah committed:
Fixed double header issue (#23895)

Vaibhavi-Mishra committed:
[Hub Generated] Publish private branch 'vaibmishra/onboardStorageResource' (#23901)

* onboarding storage resource

* addressing failures from automated checks

* onboarding as a new resource provider instead of resoruce type

* addressing the checks

* addressign check fails x2

* addressing check fails x3

* resource onboarding in the same rp

* checks fail x1

* prettier fix

* swagger model validation check

* fix the pattern change

* support put, list, patch, delete api calls

* address checks x1

* put and delete api with 2xx code

* addressing check and restructure abmsi

* fixing checks again

* checks againnnn

* fixing PAYG and adding 201 for put

* try to fix lintdiff check

* check fails

* missed this file

* fixing the error response

* fixing the error response for bmi

sasank-chaganty committed:
removing API version (#23821)

* removing API version

* removing references to 2023-03-15-preview

haolingdong-msft committed:
Create readme.java.md for SelfHelp (#23950)

robertklee commented:
Within the SchemaField definition, I think this should be x-nullable: true

![image](https://github.com/Azure/azure-rest-api-specs/assets/32447910/417c7d70-a2a2-4f9a-ac43-99f460d90202)

robertklee reviewed (changes requested):
i spotted 2 more changes

mikeharder committed:
TypeSpec CI Pipeline (#23944)

- Validates all TypeSpec specs in the repo

mikeharder committed:
[TypeSpec CI] Remove HealthInsights folders not emitting autorest (#23964)

markcowl committed:
[TypeSpec-Clean]: Content-safety changes (#23965)

* Content-safety changes

* Include swagger

* Responding to review comments

markcowl committed:
[TypeSpec-Clean]: Updating HealthInsights (#23966)

* Updating HealthInsights

* Apply suggestions from code review

Respondign to review comments

Co-authored-by: Mike Harder <mharder@microsoft.com>

* Fixing bad service name in config

---------

Co-authored-by: Mike Harder <mharder@microsoft.com>

markcowl committed:
[TypeSpec-Clean] Update for Azure.AI.TextTranslation (#23971)

* Update for Azure.AI.TextTranslation

* Update tspconfig

Co-authored-by: Mike Harder <mharder@microsoft.com>

---------

Co-authored-by: Mike Harder <mharder@microsoft.com>

markcowl committed:
Updates for OpenAi.Inference (#23970)

thomasp98296 committed:
Add new test notification APIs for Tenant Action Groups  (#23714)

* Initial changes/baseline

* New changes for 2023-05-01-preview

* Make the new api default

* Fix build

* Add regex name validation

* name validation

* Regex

* Change type.

* Fix typo

* Change to v2

---------

Co-authored-by: Thomas Pham <thompham@microsoft.com>

kazrael2119 committed:
update mysql readme.typescript (#23989)

avanigupta committed:
Update mentionees for AppConfig service (#23957)

Update mentionees for AppConfig service

Careyjmac committed:
Fix vectorSearchConfiguration x-nullable to be true

Careyjmac committed:
Fix vectorSearchConfiguration x-nullable to be true

maririos committed:
Update (#23954)

IshitaAsthana committed:
excludeDuplicateEvents query False (#23572)

Co-authored-by: IshitaAsthana <iasthana@microsoft.com>

robertklee reviewed (approved):
lgtm

maririos committed:
update link (#23994)

markcowl committed:
[TypeSpec-Clean]: Updates for AnomalyDetector (#23972)

* Updates for AnomalyDetector

* Updating config to match template

FumingZhang committed:
[AutoSyncSwagger] e97d3465c81 Merged PR 8064558: [API Release] Cherry-pick swagger changes from master branch to release branch for 2023-04 API (#23851)

Co-authored-by: apispec-automation <swagger@microsoft.com>

ahamad-MS committed:
Azure EventGrid 2023-06-01-preview: Management Swagger and Examples (#23981)

* Use 2021-10-15-preview as base-line in first commit per ARM/Swagger review requirement

* EventGrid: 2023-06-01-preview --- Add EventGrid.Json Swagger file and related examples for API Version: 2023-06-01-preview

* fix swagger by removing extra deprecated resources and their related examples

* fix prettier

* update example to trigger execution of github tasks

* fix DefaultErrorResponseSchema for caCertificates to fix some of Swagger LintDiff issues

* fix more lint diff issues

* Fix more lint diff errors

* fix lintdiff for default response

* more diffdiff fixes

* fix typo, prettier check and remove depcrecated properties

* Add required = true for discrementator

* fixed  = true for discrementator

* fix required location for discriminator for Filter

MattEberhart-MSFTE committed:
[Hub Generated] Review request for Microsoft.CostManagement to add version stable/2023-03-01 (#23892)

* Adds base for updating Microsoft.CostManagement from version stable/2022-10-01 to version 2023-03-01

* Updates readme

* Updates API version in new specs and examples

* Matteberhart/merging main for readme (#23886)

* Adds base for updating Microsoft.CostManagement from version stable/2022-10-01 to version 2023-03-01

* Updates readme

* Updates API version in new specs and examples

* Benefit Summaries Async (#22497)

* Added benefit utilizaiton summaries APIs to costmanagement.benefits.json and appropriate examples.

* Added report schema and secondary url

* Prettier check and spelling error fix

* Addressed BenefitKind / Kind in the csv schema and changed parameter name to benefitAsyncOperationIdParameter

* Replacing included quantity

* Alphabetized the parameters

* Changing request body BenefitKind property to Kind

* Adding 200 repsonse to generate api and fixing linter warnings

* Adding examples to generate api

* Making status optional for running scenario

* Changing to completed to have all properties in generate examples and adding long running operation tag to error response

* LRO Error response

* LRO lint diff error still present.

* Found all diffs in 2022-10-01 version added since branching (#23590)

* Found all diffs in 2022-10-01 version added since branching

* Fixing examples ruined by updating. Pulled straight from 2022-10-01

* Changed enum name (#23605)

* Changing code of ErrorDetails object back to a string (#23734)

* cost details report operation results example needs string error code (#23742)

* Cherry picked commit causing conflict in readme (#23817)

Co-authored-by: alprosms <92699124+alprosms@users.noreply.github.com>

---------

Co-authored-by: alprosms <92699124+alprosms@users.noreply.github.com>

* Adding x-ms-secret to the report urls (#23934)

* Remove operation results (#23938)

* Remove operation results

* Removing examples

* Using commony type error response (#23955)

---------

Co-authored-by: alprosms <92699124+alprosms@users.noreply.github.com>

Commit:
[AutoSync] 813dcc0148 Add property to override spark config for synapse notebook activity (#23903)

Co-authored-by: swagger-automation <swagger@microsoft.com>

Alancere committed:
[Go] resourcehealth add directive (#24007)

mikeharder committed:
[TypeSpec CI] Add CI and PR triggers (#24019)

markcowl committed:
[TypeSpec-Clean] Update  OpenAI.Authoring (#23969)

* Changes for OpenAI.Authoring

* Removing extraneous file

* Removing TypeSpec files for OpenAI.Authoring

mikeharder committed:
Remove folder OpenAI.Authoring (#24023)

- Spec removed from repo in #23969

mvvsubbu committed:
[Hub Generated] Review request for Microsoft.Databricks to add version stable/2023-05-01 (#23470)

* Adds base for updating Microsoft.Databricks from version preview/2022-10-01-preview to version 2023-05-01

* Updates readme

* Updates API version in new specs and examples

* Updated the readme file

* Updating missing single quotes

* Updating tag

markcowl committed:
[TypeSpec Update]: Update Versions and Anomaly Detector (#23973)

* Updates for AnomalyDetector

* Update dependencies

* Updates for AnomalyDetector

* Updating config to match template

markcowl committed:
[TypeSpec Update] Update versions and HealthInsights (#23975)

* Update dependencies

* Updating HealthInsights

* Fixing warnings and client tsp

markcowl committed:
[TypeSpec Update]: Update versions and AzureAi.TextTranslation (#23979)

* Update dependencies

* Update Azure.AI.TextTranslation

* Updating config

markcowl committed:
[TypeSpec Update]: Update Versions and ContentSafety (#23974)

* Update dependencies

* Updates for ContentSafety

* Update for final operation change

bavneetsingh16 committed:
Microsoft.KubernetesConfiguration to add preview and stable version 2023-05-01 (#23676)

* copy stable API version from 2022-11-01 to 2023-05-01 and preview API 2022-01-15-preview to 2023-05-01-preview

* updating the API version to 2023-05-01

* updating the API version to 2023-05-01-preview

* update readme files to fix avocado errors

* update readme file to remove operations from the list

* fix model validation errors

* fix operations issue in lint errors

* fix swagger lintdiff errors

* fix pattern

* add nextLink property

* add pattern restriction fix model validation errors

* fix the path for common file

* fix readme and x-ms-identifiers errors

* fix nextLink error

* resolve changes suggested in comments

* create sdk for preview api version

* add support for patch operation

* Update readme.python.md

---------

Co-authored-by: Bavneet Singh <bavneetsingh@microsoft.com>
Co-authored-by: Yuchao Yan <yuchaoyan@microsoft.com>

kjilla committed:
Updating the name of the Storage Account Type for Target Extended Locations in Gallery.json (#23741)

Commit:
add community gallery image identifier (#23820)

Co-authored-by: Theodore Chang <theodore.l.chang@gmail.com>

Jingshu923 committed:
[AutoSync] 09bc56fa85 Add outputColumns x-ms-format-element-type Property (#24003)

Co-authored-by: swagger-automation <swagger@microsoft.com>

serbrech committed:
[Hub Generated] Publish private branch 'containerservice/official/fleet-api-release' (#23947)

* [AutoSyncTsp] da7cce64aed Merged PR 8041779: [Fleet] update spec release to include typespec files

* [AutoSyncSwagger] da7cce64aed Merged PR 8041779: [Fleet] update spec release to include typespec files

---------

Co-authored-by: apispec-automation <swagger@microsoft.com>

Jingshu923 committed:
[Hub Generated] Publish private branch 'datafactory/v-jinpan/SqlFix' (#23985)

* [AutoSync] 1066864ec6 Fix Sql Family Missing Property isolationLevel

* [AutoSync] 272b1fd4e8 update

---------

Co-authored-by: swagger-automation <swagger@microsoft.com>

kazrael2119 committed:
update sample parameter (#24027)

mikeharder committed:
[TypeSpec CI] Fix typo in triggers (#24042)

arthurwang2021 committed:
[Hub Generated] Review request for Microsoft.Help to add version stable/2023-06-01 (#23956)

* Adds base for updating Microsoft.Help from version preview/2023-01-01-preview to version 2023-06-01

* Updates readme

* Updates API version in new specs and examples

* Added additional status code for azure powershell cmdlet.

* Updated the description for 200&201 status code.

* Rephrase the description.

eatkimchee committed:
[Hub Generated] Publish private branch 'eatkimchee-fist-Microsoft.IoTFirmwareDefense-2023-02-08-preview' (#23921)

* add or modify files

* Fix validation errors

- remove references to old private preview versions from readme.md
- add binary hardening flags to custom-words.txt

* make changes from review comments

- remove unused param
- rename definition to match usage
- make sas token response properties more clear with description and format and readOnly
- move domain-specific terms to cSpell.json

* fix urls in examples

* remove x-nullable from string url properties

---------

Co-authored-by: rflmk <mkennedy@refirmlabs.com>

mikaelmello committed:
[AutoSync] 74754a3d25 Update properties (#23910)

Co-authored-by: swagger-automation <swagger@microsoft.com>

vatsalyachauhan committed:
[Hub Generated] Publish private branch 'dev-dns-Microsoft.Network-2023-07-01-preview' (#24037)

* Intermediate Commit - Added DnssecConfigs resource type

* Add RelativeRecordSetNameParameter to remove duplicate configs

* Add RecordTypeParameter to remove duplicate configs

* Revert "Add RecordTypeParameter to remove duplicate configs"

This reverts commit f97bd65a09ee611d77c9c6080ecc4cac2c296a01.

* Add Definitions for new record types

* Add placeholders for tests

* Add examples for dnssecconfigs

* Reference Patch examples for DS, TLSA and NAPTR

* Add delete api examples for ds, naptr and tlsa recordsets

* Add examples for NAPTR recordset

* Add examples for DS and TLSA recordset

* Add 2023-07-preview version for dns in Autorest configs

* Fix API version in examples

* Address comments

* Fix Lint errors

* Fix model validation

* Fix model validation and delete privatedns examples

* Fix UNREFERENCED_JSON_FILE

* Fix SpellCheck

* Fix MissingTypeObject

* Fix Validations

* Fix XmsIdentifierValidation

* Fix PathForNestedResource and PathForNestedResource

* Fix PageableOperation and LroLocationHeader

* Remove systemData from Dnssec

* Fix RequiredReadOnlySystemData

* Fix XmsParameterLocation

* Fix XmsIdentifierValidation

* Fix ParametersOrder

* Fix PathForNestedResource

* Fix indentation

* Check

* Fix XmsLongRunningOperationOptions TopLevelResourcesListBySubscription OperationsAPIImplementation

* Fix nextlink for list dnssecconfigs

* Fix ParameterNotUsingCommonTypes

* Fix some validations

* Fix

* Refactor RecordTypeParameter

* Revert resource type references from common-types

* Suppress ResourceMustReferenceCommonTypes and LroErrorContent

* Suppress GetCollectionOnlyHasValueAndNextLink

* None

* Use new suppression format

* Suppress OperationsAPIImplementation and GetCollectionOnlyHasValueAndNextLink without where condition

* Fix descriptions

* Fix AvoidAdditionalProperties

* None

* Address comments

* Add DNSKEY and DDDS in custom-words list

* Address comments

* Address comments

* Improve suppression docs

* Change Dns to DNS in description

* False commit to kick git CI

---------

Co-authored-by: Vatsalya Chauhan <vachauha@microsoft.com>

msyyc committed:
Update readme.python.md (#24047)

ericasp16 committed:
Communication/PhoneNumbers - Add Number Lookup API (#23053)

* add new 2023-05-01-preview version to communication-phone-numbers

* add number lookup API definitions and examples

* add tag to readme

* fix references to CommunicationErrorResponse

* fix version numbers, problems with examples

* fix model validation

* update with feedback from ACS review

* temporarily removing 2023-05-01-preview in order to base it off of the latest preview version, not the stable version

* re-adding 20203-05-01-preview, copied from 2022-06-01-preview

* add operator information search endpoint and example

* update examples

* updated to fix automatic validation

* fix model validation error

* temporarily remove 2023-05-01-preview to base off newest version

* re-create 2023-05-01-preview, based off latest stable version

* update 2023-05-01-preview with number lookup endpoint information

* fix model validation

* attempt to fix a naming conflict

* update documentation comments, operationId

* fix example to match response in swagger

* revert to 'Error' name

* fix lintdiff

mssfang committed:
[OpenAI] Enable sync stack and not generate tests folder in tspconfig.yaml (#24033)

* enable sync stack and not generate tests folder in tspconfig.yaml

* Update specification/cognitiveservices/OpenAI.Inference/tspconfig.yaml

ericasp16 committed:
update values of OperatorNumberType enum (#24052)

dzeliar committed:
SIP-Update formatting of security section (#23952)

* update formatting of security section to inlcude both authentication in generated documentation

* rename Authentication to access_key to be consistent in generated documentation

deyaaeldeen committed:
[OpenAI] Update properties back from optional to nullable (#24039)

yanjungao718 committed:
modify type for WebActivityAuthentication-resource property (#23949)

abhinav-ghai committed:
Add data-plane 2023-06-30 for Azure Digital Twins (#23616)

* Add data-plane 2023-06-30 for Azure Digital Twins

* Add x-ms-error-code to header

Commit:
Improve message for the NewApiVersionRequired label (#24058)

GalGoldi72 committed:
Add tag for dotnet sdk generation (#24016)

* Add tag for dotnet sdk generation

* Fix tag

xiaoxuqi-ms committed:
Add 2022-09-30-preview for service operations in MySQL (#23900)

* add base by copying from 2021-12-01-preview

* update api version

* update readme.md

* use common types

ambrahma committed:
[PostgreSQL] Allow network property to be updated during PG flexible server update (#23929)

* Allow entwork property to be updated during PG flexible server update

* Updated description to explicitly specify when delegatedSubnetResourceId and privateDnsZoneArmResourceId are required

dsmith111 committed:
Update/Fix tsc package name (#24061)

jeremyfrosti committed:
Release sql microsoft.sql 2022 11 01 preview (#23802)

* Adds base for updating Microsoft.Sql from version preview/2022-08-01-preview to version 2022-11-01-preview

* Updates readme

* Updates API version in new specs and examples

* Add Swagger API changes required for progress-reporting related changes in DatabaseOperations (#22669)

* Update swagger (#22715)

* Follow-up Swagger API changes required for progress-reporting related changes in DatabaseOperations  (#22775)

* Add Swagger API changes required for progress-reporting related changes in DatabaseOperations

* Follow up changes required for progress-reporting related changes in DatabaseOperations

* updated example file

* Added Failed case in list-operations examples

* Update Microsoft.Sql ManagedInstance 2022-11-01-preview Swagger Docs (#22899)

* Update Microsoft.sql ManagedInstance 2022-11-01-preview Swagger Docs

* Address Swagger Validation Errors

* Modify custom-words.txt

* Manually updating arm-id format to match stable 2021-11-01 version

* Adding provisioning state changes

* fixing validation

* adding missing space

* Remove HTTP 200 from Start/Stop Actions as we only return HTTP 202

---------

Co-authored-by: Jovana Toroman <jtoroman@microsoft.com>

* Update DateTime to ISO-8601 format in example file (#22986)

* Add Swagger API changes required for progress-reporting related changes in DatabaseOperations

* Follow up changes required for progress-reporting related changes in DatabaseOperations

* updated example file

* Added Failed case in list-operations examples

* Update DateTime to ISo-8601 format in example file

* Enable TryPlannedBeforeForcedFailover for SQL DB Failover Group (#22673)

* Enable TryPlannedBeforeForcedFailover for SQL DB Failover Group

* Modify example file

* Add example

* Fix ModelValidation issues

* Fix ModelValidation issues

* Add new controller

* Modify example

* Modify example

* Update server name

* Fix LintDiff

* Fix model validation errors

* Fix PostOperationAsyncResponseValidation, Unfix LroLocationHeader,ResourceNameRestriction

* Fix PostOperationAsyncResponseValidation

* Fix pending items

* Update doc

* Remove empty/unschematized object

---------

Co-authored-by: subhrob <subhrob@microsoft.com>

* [DO NOT MERGE] Add Archive option to LTR Policy for SQL DB (#22697)

* add archival changes to LTRPolicy

* fix LongTermRetentionPolicies swagger file

* fix check failures

* redo swagger

* Update swagger documentation for Managed Instance start-stop actions (#22992)

* Add NetworkSecurityPerimeterConfigurations Swagger (#22960)

* Add NetworkSecurityPerimeterConfigurations Swagger

* Fixes round 1

* Fixes round 2

* Fixes Round 3

* Fixes Round 4

* Fix Round 5

* fixing enum (#22995)

Co-authored-by: Jovana Toroman <jtoroman@microsoft.com>

* add changes (#23120)

* Carry all minor changes to dev sql microsoft.sql 2022 11 01 preview (#23337)

* Revert "add changes (#23120)"

This reverts commit 5c348d22f8fb6c29b7322e28ea78d665586f4135.

Revert "[DO NOT MERGE] Add Archive option to LTR Policy for SQL DB (#22697)"

This reverts commit 559f7f474d944a92f6cb108eb433a05f3c53cd8e.

* Carrying minor changes to release branch for Microsoft.Sql 2022-11-01-preview

* Reverting DatabaseAdvisors

* Further reverting problematic files

* Reverting a few example files

* Adding back arm-id to a few files

* Reverting error message change in ManagedDatabaseRestoreDetails.json, Jobs.json,ServerDevOpsAudit.json, and TransparentDataEncryptions.json

* Minor fixes for Turkey name change and missing x-ms-identifier (#23637)

* Fixing issues with a few breaking changes (#23943)

* Minor fixes for Turkey name change and missing x-ms-identifier

* Adding back missing arm-id and other missing property issues causing breaking changes incorrectly.

* Fixing prettier issue with ManagedDatabases.json

* Fixing arm-id format in 4 examples

* Adding missing location for 202 responses on sync examples

---------

Co-authored-by: Ji Wang <wangjiprc@gmail.com>
Co-authored-by: Mohit Agarwal <mohitagarwal@microsoft.com>
Co-authored-by: GalGoldi72 <124780962+GalGoldi72@users.noreply.github.com>
Co-authored-by: ilijadivljan-ms <122982104+ilijadivljan-ms@users.noreply.github.com>
Co-authored-by: Jovana Toroman <jtoroman@microsoft.com>
Co-authored-by: subhrob <subhro.bhattacharya@gmail.com>
Co-authored-by: subhrob <subhrob@microsoft.com>
Co-authored-by: CelinaJiangJXY <124339390+CelinaJiangJXY@users.noreply.github.com>
Co-authored-by: beoberha <51964155+beoberha@users.noreply.github.com>
Co-authored-by: toki95 <35834415+toki95@users.noreply.github.com>

SrinikhilReddy committed:
[Hub Generated] Review request for Microsoft.DocumentDB to add version preview/2023-03-15-preview (#23327)

* Adds base for updating Microsoft.DocumentDB from version preview/2023-03-01-preview to version 2023-03-15-preview

* Updates readme

* Updates API version in new specs and examples

* combined changes for managed cassandra preview swagger.

* materializedview swagger changes

* changing preview release version

* Reverting changes to 2023-03-01-preview

* preview changes for nova.

* Pretty check.

* Add files via upload

* Update cosmos-db.json

adds database partition merge api.

* saving work

* saving work

* saving work

* Add example for swagger apis.

* fix cluster name and rg values in example.

* Add cassandraProcessStatus property.

* format example.

* Fixing model validation errors.

* Resolving model validation errors.

* resolve Lint Diff issues.

* removing duplicate entry for package-preview-2023-03

* Fixing spelling.

* Fix pretty validation.

* try removing modelerfour:

* update readme to 2023-01-15-preview

* bring back both 2023-03-01 and 2023-03-15 preview in readme.md

* update tag to generate for  package-preview-2023-03-15

* ring back CassandraClusterRepairPublicProperties.

* merge conflict resolution

* port listconnectionstrings changes from 03-15-stable

* backport stable to preview swagger.

* port location api updates from 3-15-stable to preview version

* Fix model validation error :-).

* remove duplicate status property

* Undo Vivek's changes.

* saving work

* saving work

* local changes

* local changes

* local changes

* local changes

* temporally resolve duplicate schema issue

---------

Co-authored-by: Vivek Sundararajan <visunda@microsoft.com>
Co-authored-by: Chandrasekhar Gunturi <chguntur@microsoft.com>
Co-authored-by: Carolyn Jackson <carjackson@microsoft.com>
Co-authored-by: Chenjie Shi <tadelesh.shi@live.cn>

thang-bit committed:
[Defender (RiskIQ) - EASM] Api version 2023-03-01-preview (#23147)

* Added new version that include data connector api.

* Update easm api version to include data connection

* Update readme

* Update version in readme, rename example DataConnections_Remove.json

* Address comments on easm api 2023-03-01-preview version

* Update List operations examples and add Assets_Observations example

* Update model of examples

* Update data connections examples

* Remove properties field from DataConnections_Get example

* add discriminator to data connection base classes

* Add properties to examples

* Fix typos

* Remove assets observations endpoint and its example

* the `value` property of a pageable responses is required

---------

Co-authored-by: Brian Zak <brianzak@microsoft.com>

kazrael2119 committed:
update-network-directive (#24067)

RheaHooda23 committed:
Adding new version 2023-05-01-preview for adding an additional parameter in "MarketplaceSaaSInfo" response (#24031)

* Base API Commit

* Changing API version to 2023-05-01-preview

* adding additional parameter "marketplaceStatus"

* updating readme.md

Jingshu923 committed:
[Hub Generated] Publish private branch 'datafactory/v-jinpan/FixO365' (#24085)

* [AutoSync] 4844c5088f Add OutputColumnsElement For Track2 Upgrade

* [AutoSync] 0fe3e1a701 update

* [AutoSync] 39056bdca3 update

---------

Co-authored-by: swagger-automation <swagger@microsoft.com>

karthikku-2020 committed:
ViewPropertiesUpdate (#24020)

tleibovitz committed:
User solutions authenticated API (#23585)

* first pr

* prettier fix

* fix ref

* response code

* readme

* readme api version

* api version

* try fix avocado

* try again

* another one

* try to remove default tag

* semantic validation

* operations

* autorest

* a few more fixes to examples

* save progress

* fix all issues

* custom word

* comments

* ARM review comments

* remove unnecessary files

* remove local changes

* minor modification

ambrahma committed:
[Hub Generated] Review request for Microsoft.DBforPostgreSQL to add version preview/2023-03-01-preview (#23402)

* Adds base for updating Microsoft.DBforPostgreSQL from version stable/2022-12-01 to version 2023-03-01-preview

* Updates readme

* Updates API version in new specs and examples

* Update api and add examples to support CMK with geo-backup

* Fix object missing required property error for userassignedidentity

* Fix multiple default tags error in readme file

* Update Typescript and Python readme files to include new api version 2023-03-01-preview

* Add storage autoGrow support for api version 2023-12-01

* Add support for location capabilities and server capabilities api for api version 2023-03-01-preview

* FIx some swagger linter errors

* Fix linter issues and fix the response for Location capabilities

* Fix linter and prettier errors for capabilities api

* Fix linter error

* Fix linter error for missing identifier id

* Add support for IOPS scaling properties for PG flexible servers

* Remove unsupported type "SystemAssigned" for User AssignedIdentities after syncing code with main branch

* Fix review comment to have iopsTier as optional set, however both iopsTier and iops should be part of response.

* FIx the object for capabilities api to be in sync with the RP

* Update boolean properties to string instead for Enabled/Disabled

* Fix examples for capabilities api

* Add an entry for LogFiles controller to list server logs for PostgreSQL flexible server

* Added an entry for ServerLogs in readme.md file

* Add new supported property, onlineResizeSupported for capabilities api

* Fix path for common types.json

* Fix paths for common types.json to fix validation errors

* DIrectly refer the common type ErrorResponse definition for default response for logFiles api

* Add advancedThreatProtectionSettings api and corresponding samples to 2023-03-01-preview api version for PostgreSQL flexible servers

* Add threat protection to name tag for api version 2023-03-01-preview

* Add migrations api and migrationCheckNameAvailability api to 2023-03-01-preview version

* Fix linter errors for ThreatProtection

* Fix linter errors for Migrations

* Fix linter errors for capabiities

* Update boolean properties for Migrations to enum

* Add long term retention APIs to 2023-03-01 preview

* Fix the request and response based on controller actions for long term retention backup apis

* Fix lintdiff warnings

* Fix review comments

* Fix issue related to tools like credscan, spellcheck, api swagger view

* Fix linter errors

* Revert changes for Update for Migrations api

* Fix file path for Migrations

* Fix pyton track errors

* Fix typo for Migrations properties

* Updated description for review comments

* Fix typo

* Fix SDK automation errors

* Fixed review comment

* Updated api name for post operation and updated descriptions

* Add missing body parameters for threat protection api

* Fix linDiff error to add update in PUT description for Migrations api

* Remove unwanted parameters for long term retention apis

* Add support for ReviveDropped createMode for 2023-03-01-preview api version

* Fix file reference

* Fix S360 issue for sourceServerResourceId in response and fix descriptions for other issues

* Fix S360 for replicaCapacity returned only in response but not set in PUT

* Remove newly added Server Threat Protection api from swagger for api version 2023-03-01-preview

---------

Co-authored-by: Tanay Varma <tavarma@microsoft.com>

yuc-Li committed:
suppress 2023-04-15-preview swagger linter rule R3016 (#24084)

Co-authored-by: v-yuchenli <v-yuchenli@microsoft.com>

subhrob committed:
Adding Failover Groups API for SQL DB on v5 tag for 2022-11-01 (#24069)

Co-authored-by: subhrob <subhrob@microsoft.com>

jingzhong-MS committed:
create 2023-04-18-preview to add actionOnExistingTargetTable property (#23670)

* remove dropExistingTargetTableOnStart properties

* add actionOnExistingTargetTable properties

Co-authored-by: jingjingzhong123 <131136533+jingjingzhong123@users.noreply.github.com>

l0lawrence committed:
EventGridv2 TypeSpec Api Preview (#23204)

* start typespec

* adding eventgrid typespec for api w/ TODOs

* update eventgrid typespec with latest eventgrid v2 operations

* don't require content-type if there is no body

* Update specification/eventgrid/typespec/main.tsp

Co-authored-by: JoshLove-msft <54595583+JoshLove-msft@users.noreply.github.com>

* Update specification/eventgrid/typespec/main.tsp

Co-authored-by: JoshLove-msft <54595583+JoshLove-msft@users.noreply.github.com>

* changing naming of cloudevent and added in data_base64

* openapi.json

* Update specification/eventgrid/typespec/main.tsp

Co-authored-by: Libba Lawrence <llawrence@microsoft.com>

* Update specification/eventgrid/typespec/main.tsp

Co-authored-by: Libba Lawrence <llawrence@microsoft.com>

* Update specification/eventgrid/typespec/main.tsp

Co-authored-by: Libba Lawrence <llawrence@microsoft.com>

* lockTokens format, updated json, optional? params

* address code review comments

* name_change

* add @internal for python

* Update specification/eventgrid/Azure.Messaging.EventGrid/main.tsp

Co-authored-by: JoshLove-msft <54595583+JoshLove-msft@users.noreply.github.com>

* move @internal to client.tsp (#23538)

* rename (#23565)

* [EventGrid Typespec] breaking changes with april release of typespec (#23539)

* breaking changes with april release of typespec

* unknown type

* [EG Typespec] Update Release behavior (#23699)

* update behavior

* just behavior

* Add tspconfig and remove AAD auth (#23717)

* add tspconfig

* add namespace

* remove oauth

* [EG TypeSpec] Archboard Comments (#23696)

* refactoring off of apiview

* keep as int

* no duration

* aligning ack and release

* remove behavioral change

* ack to release

* initial naming changes

* Update ReleaseResult doc

Co-authored-by: JoshLove-msft <54595583+JoshLove-msft@users.noreply.github.com>

* Update AckResult doc

Co-authored-by: JoshLove-msft <54595583+JoshLove-msft@users.noreply.github.com>

* versioning twice-- remove one instance

---------

Co-authored-by: JoshLove-msft <54595583+JoshLove-msft@users.noreply.github.com>

* Address couple of stewardship team feedback. These include: 1. Rename CloudEventEvent to simply CloudEvent, 2. Add more description to the operations including the possible erorr codes, 3. Add PublishResult with empty Json object as successful response for the Publish operation, 4. Others.

* Add support for missing Reject operation + adding deliveryAttemptCount to BrokerProperties + Adding query parameter for release operation for deliveryDelayInSeconds

* Update failedTokens/SuccessfulTokens Description to address code review comments

* Update to match service behavior (#23754)

* Update to match service behavior

* remove locktoken

* [EGv2] Editing unused variables (#23917)

* event delivery delay not in preview

* remove from url comment

* [EGv2] Version dependency on Azure.Core  (#23936)

* verioning fix

* spacing mishap?

* [EventGrid] Deliveryattempt change (#23960)

* deliveryCount 5/1

* small typo

* [EventGrid] Remove internal (#23995)

* remove internal

* remove client.tsp

* remove waitWaitTime (#24078)

* move location of json file (#24076)

* [Egv2] Encode param (#24080)

* encode

* remove num default on duration

---------

Co-authored-by: Laurent Mazuel <laurent.mazuel@gmail.com>

* [EGv2] Fix pipeline (#24098)

* regen off new commit for encode

* reference preview tag

* ignore word

* update readme to have both apis

* update with next autorest

* change format to int32

---------

Co-authored-by: Ashraf Hamad <ahamad@ntdev.microsoft.com>
Co-authored-by: Laurent Mazuel <laurent.mazuel@gmail.com>
Co-authored-by: JoshLove-msft <54595583+JoshLove-msft@users.noreply.github.com>
Co-authored-by: Ashraf Hamad <ahamad-MS@users.noreply.github.com>

mikeharder committed:
Update typespec-autorest and typespec-azure-core (#24100)

- Includes fix for Azure/typespec-azure#3024

us6193 committed:
Adding missing customerName property to fix S360 swagger correctness (#23948)

Co-authored-by: Umang Shah <umangshah@microsoft.com>

tianxin-ms committed:
Tianxinliu/20230501 (#23661)

* Adds base for updating Microsoft.CognitiveServices from version stable/2022-12-01 to version 2023-05-01

* Updates readme

* Updates API version in new specs and examples

* Cognitive Services 2023-05-01 56f941482c7ed5e329cc96d4f23b3d0a35ad31a6
author Tianxin Liu <tianxinliu@microsoft.com> 1681797091 -0700
committer Tianxin Liu <tianxinliu@microsoft.com> 1683320

Cognitive Services 2023-05-01

* fix follow comments

* fix CI issues

* remove Rai related from 0501 stable

* add format for datetime

* fix a typo

* undo "remove callratelimit" to fix breaking change

---------

Co-authored-by: Andrew Yang <yangyuan@users.noreply.github.com>

yirsu committed:
Add "Warning" to the EP HealthStatus field (#23942)

* Add EP healthStatus property

* Fix Private Endpoint swagger issue

* Remove 202 response from Private Endpoint Connection model

* Add warning status for ep health

kevinmichaelbowersox committed:
Purview Share Test Api Specification Update - Version 2023-05-30-preview (#23993)

* changes

* changes

* changes

* changes

* changes

* changes

---------

Co-authored-by: faisalaltell12 <faisalaltell@gmail.com>

weidongxu-microsoft committed:
mgmt, resources subscriptions, revert SubscriptionIdParameter (#24088)

Careyjmac committed:
Add vectorIndexSize to index and service statistics

Careyjmac committed:
Merge branch 'polyFix' of https://github.com/Careyjmac/azure-rest-api-specs into polyFix

Careyjmac commented:
Here is the new PR for prosperity: #24112
